runtime/pkg/rduckdb/db.go

```diff
@@ -725,15 +734,22 @@ func (d *db) openDBAndAttach(ctx context.Context, uri, ignoreTable string, read
 	// this is required since spaces and other special characters are valid in db file path but invalid and hence encoded in URL
 	connector, err := duckdb.NewConnector(generateDSN(dsn.Path, query.Encode()), func(execer driver.ExecerContext) error {
 		for _, qry := range d.opts.InitQueries {
-			_, err := execer.ExecContext(context.Background(), qry, nil)
+			_, err := execer.ExecContext(ctx, qry, nil)
 			if err != nil && strings.Contains(err.Error(), "Failed to download extension") {
 				// Retry using another mirror. Based on: https://github.com/duckdb/duckdb/issues/9378
-				_, err = execer.ExecContext(context.Background(), qry+" FROM 'http://nightly-extensions.duckdb.org'", nil)
+				_, err = execer.ExecContext(ctx, qry+" FROM 'http://nightly-extensions.duckdb.org'", nil)
 			}
 			if err != nil {
 				return err
 			}
 		}
+		if !read {
+			// disable any more configuration changes on the write handle via init queries
+			_, err = execer.ExecContext(ctx, "SET lock_configuration TO true", nil)
```

Comment:
Does this prevent the `afterCreateFn` from cleaning up e.g. a secret?

Reply:
No it does not. The following chain of queries succeed:
```
CREATE SECRET secret...
SET lock_configuration TO true;
DROP SECRET secret;
```